libs/core/src/main/java/org/opensearch/core/common/Strings.java

```diff
@@ -815,4 +815,17 @@ public static String toLowercaseAscii(String in) {
         }
         return out.toString();
     }
+
+    /**
+     * Check whether every single character in the string is a digit.
+     *
+     * Important: this also returns {@code true} for empty strings!
+     *
+     * @param s the string, must not be null.
+     * @return true if the string is empty or only contains digits, false otherwise.
+     */
+    public static boolean isDigits(final String s) {
```

Comment:
> * Important: this also returns {@code true} for empty strings!

I think this is very surprising  behaviour  and should be revisited, the empty string is not valid set of digits.